OpenRA.Platforms.Default/ThreadedGraphicsContext.cs

```diff
@@ -92,8 +90,12 @@ void RenderThread(object contextObject)
 							return new ThreadedFrameBuffer(this,
 								context.CreateFrameBuffer(t.Item1, (ITextureInternal)CreateTexture(), t.Item2));
 						};
-					getCreateShader = name => new ThreadedShader(this, context.CreateShader((string)name));
-					getCreateVertexBuffer = length => new ThreadedVertexBuffer(this, context.CreateVertexBuffer((int)length));
+					getCreateShader = bindings => new ThreadedShader(this, context.CreateShader((IShaderBindings)bindings));
+					getCreateVertexBuffer = (t, type) =>
+					{
+						var vertexBuffer = context.GetType().GetMethod("CreateVertexBuffer").MakeGenericMethod(type).Invoke(context, new[] { t });
+						return typeof(ThreadedVertexBuffer<>).MakeGenericType(type).GetConstructors()[0].Invoke(new[] { this, vertexBuffer });
```

Comment:
IIRC, can `GetConstructors()[0]` become just `GetConstructor()`?

Reply:
It requires parameters to be passed in which add more reflection